nameko/messaging.py

```diff
@@ -149,15 +148,16 @@ def retry_policy(self):
         return DEFAULT_RETRY_POLICY
 
     def setup(self):
-
         exchange = self.exchange
-        queue = self.queue
+        queues_to_declare = self.queues_to_declare
 
         verify_amqp_uri(self.amqp_uri)
 
         with get_connection(self.amqp_uri) as conn:
-            if queue is not None:
-                maybe_declare(queue, conn)
+            if queues_to_declare is not None:
+                for queue in queues_to_declare:
+                    if queue is not None:
+                        maybe_declare(queue, conn)
```

Comment:
It turns out that `producer.publish` accepts a `declare` kwarg, that will do this for you. It might be easier to test within nameko, but you could drop these declarations and allow kombu to do it. Up to you.